packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -456,44 +456,44 @@ export async function getAvailableSlots({ input, ctx }: GetScheduleOptions): Pro
     });
   }
 
+  const users = usersWithCredentials.map((currentUser) => {
+    return {
+      ...currentUser,
+      currentBookings: currentBookingsAllUsers
```

Comment:
I wonder if we can use a reduce here to make this even faster - we are filtering and then mapping which is 2 itterations - i think we can make it just one with something like 
(Note this will type error as we'd need to type the array in the reduce function and i didnt spend enough time to figure it out)
```
currentBookingsAllUsers.reduce((accumulator, booking) => {
  if (booking.userId === currentUser.id || booking.attendees?.some((a) => a.email === currentUser.email)) {
    const { attendees: _attendees, ...bookingWithoutAttendees } = booking;
    accumulator.push(bookingWithoutAttendees);
  }
  return accumulator;
}, []);
```

Reply:
I didn't do that because even with 500 users, multiple iterations won't make a noticeable difference.

Plus reduce is less readable than map.

Reply:
Yeah great call, @hariombalhara. Scalars are removed in Big-O notation. i.e. O(2n) = O(n)